File: example/moon/src/main/java/com/moon/services/CartService.java
package com.moon.services;

import com.moon.Article;
import com.moon.persistence.CartDao;

import java.util.List;

public enum CartService {
    INSTANCE;

    private final CartDao cartDao = CartDao.INSTANCE;
    public void addArticle(Article article) {
        cartDao.addArticle(article);
    }

    public void removeArticle(Article article) {
        cartDao.removeArticle(article);
    }

    public List<Article> getArticles(){
        return cartDao.getArticles();
    }

    public void clear() {
        cartDao.clear();
    }
}

File: example/sun/src/main/kotlin/com/sun/services/CartService.kt
package com.sun.services

import com.sun.models.Article
import com.sun.persistence.CartDao

object CartService {

    fun addArticle(article: Article) {
        CartDao.addArticle(article)
    }

    fun removeArticle(article: Article) {
        CartDao.removeArticle(article)
    }

    fun getArticles(): List<Article> {
        return CartDao.getArticles()
    }

    fun clear() {
        CartDao.clear()
    }
}